# Welcome to ConSERT - A brief introduction
Kia ora and welcome the the Conservation, Systematics and Evolution Research Team repository! Here, we have a brief introduction to the command line interface (CLI), data carpentry, data management, and bioinformatic tools commonly used in this group. 

This information is freely available to members of ConSERT and readily available for modification as bioinformatic approaches/practices inevitably change overtime. We're also open to and welcome feedback on how this documentation. :fire: 

It is important to note that the information provided here in this repository is simply to serve as a brief introduction to concepts and practices you'll likely use as a member of our group. It is NOT intended to serve as the only resource you would want to access. In fact, there are a wealth of resources available elsewhere that will cover this material better and with more detail than is provided here. Links to suggested resources for additional learning will be made available within each directory. :smiley:

Below is a overview of each of the directories in this repository. To get started on any particular topic, head to the relevant directory above or click on the links provided below. 

## 1) Introduction to the command line and bash
[This section](https://github.com/UC-ConSERT/Welcome-to-ConSERT/tree/main/1_Intro_to_bash) is for anyone who may be new to the command line interface (CLI), or just needs a refresher. This sections explores common commands for parsing data files, maneuvering through directories and simple loops. We also introduce some regular expressions and tips for writing scripts.

## 2) Introduction to Git
In [this section](https://github.com/UC-ConSERT/Welcome-to-ConSERT/tree/main/2_Intro_to_Git) we explore Git, what it is, why you want to use it, and how it may be helpful when working on collaborative projects.

## 3) Data Management Principles
Given that many members of our group work with taonga species, [this section](https://github.com/UC-ConSERT/Welcome-to-ConSERT/tree/main/3_Data_management) provides some FAIR and CARE principles our group operates under when handling data and scripts. We also provide recommendations for organising your own directories. 

## 4) Components and structure of compute clusters
Chances are you will be working with large data sets that require more RAM and storage than is readily available on a desktop computer. [Here](https://github.com/UC-ConSERT/Welcome-to-ConSERT/tree/main/4_Intro_to_Clusters), we briefly outline the components of a cluster (e.g., login node vs. compute nodes) and some considerations when using one. The concepts outlined in steps 1, 2, and 3 really start coming together here. :wink:

## 5) Genotyping by Sequencing quality control and preliminary analyses
If you're working with genotyping-by-sequence data (GBS), [here](https://github.com/UC-ConSERT/Welcome-to-ConSERT/tree/main/5_Intro_to_GBSdata) we provide some template scripts for running the STACKS pipeline. We outline steps for pre-processing and trimming your reads, and the next steps for either genome alignment or an alignment free approach to SNP discovery.

## 6) Whole genome resequence data quality control and preliminary analyses
If you're working with Illumina resequence data, [here](https://github.com/UC-ConSERT/Welcome-to-ConSERT/tree/main/6_Intro_to_Illumina_WGS) we provide template scripts for quality control of reads, alignment to a reference genome and quality metrics. 